client/src/js/services/FilterService.js

```diff
@@ -131,6 +131,15 @@ function FilterService() {
     return angular.copy(this._filterIndex);
   };
 
+  // uncachable filters should not remain for a long time, so they can be remove if needed
+  FilterList.prototype.removeUnCachableFilters = function removeUnCachableFilters(filters) {
+    filters.forEach(filter => {
```

Comment:
This is good code!

Instead of making a separate API call, I propose that you put this in the function `FilterService.prototype.formatCache()`.  Since that is where the cache is saved, if you put this first, they will never be cached.  Then you won't need to called `initLoading()` twice and can remove the `Journal.filters.removeUnCachableFilters($state.params.filters)` call.

Reply:
where  formatCache is called?

Reply:
On [line 151 of the JournalService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/journal/journal.service.js#L151).

To be clear, you would change the `formatCache()` function, not the `JournalService`.  Basically, the rule would be "if `uncacheable` is set, do not return this filter from the `formatCache()` function call".

Reply:
but when we have uncachable and cachable params in the filter we will have to load

Reply:
Excuse me, I meant `cacheable`.  If `cacheable === 0`, as your code shows, it shouldn't return that filter from the `formatCache()` call.

I think that if you just change the formatCache, it will not have any affect on the HTTP request .. just on the ones that are saved.